services/ark-cluster-memory/ark-cluster-memory/src/routes/memory.ts

```diff
@@ -61,8 +59,11 @@ export function createMemoryRouter(memory: MemoryStore): Router {
         return;
       }
       
-      // Store messages with full metadata
-      memory.addMessagesWithMetadata(session_id, query_id, messages);
+      // Extract agent metadata from ark object
+      const agentName = ark?.agent_name || '';
+      
+      // Store messages with metadata
+      memory.addMessagesWithMetadata(session_id, query_id, messages, undefined, agentName);
```

Comment:
It looks like `agent prompt` is never recorded, do we need or use that field? the agent prompt is probably overloading something, cause the prompt is simply the message with the role `system` for the agent name (i think)